src/cluster/registration.go

```diff
@@ -43,30 +52,48 @@ func (c DeviceCluster) DeviceDiscovery(scanDurationSeconds int) {
 // HandleDiscoveredDevice does this with some additional steps. For example:
 // 1. does the Device already exist in the DB? (MAC address is the unique identifier in this case).
 // 2. immediately check its health.
-func (c DeviceCluster) HandleDiscoveredDevice(newDevice model.NewDevice) error {
+func (c DeviceCluster) HandleDiscoveredDevice(newDevice model.NewDevice) (model.Device, error) {
 	results, err := c.DeviceRepository.Get(model.Device{
 		MAC: *newDevice.Mac,
 	})
 	if err != nil {
-		return err
+		return model.Device{}, err
 	}
 
-	if len(results) > 0 {
-		return nil
-	}
+	discoveredDevice := new(model.Device)
+	switch len(results) {
+	case 0:
+		discoveredDevice, err = c.DeviceRepository.Create(newDevice)
+		if err != nil {
+			return model.Device{}, err
+		}
+	case 1:
+		discoveredDevice := device.FromNewDevice(newDevice)
+		discoveredDevice.Active = true
 
-	completeDevice, err := c.DeviceRepository.Create(newDevice)
-	if err != nil {
-		return err
+		if err := c.DeviceRepository.Update(updateDeviceFromDevice(&discoveredDevice)); err != nil {
+			return model.Device{}, err
+		}
+	default:
+		return model.Device{}, errors.New("multiple results returned for 1 mac address")
 	}
 
-	c.logger.Debug(fmt.Sprintf("registered mac address [%s] with id [%s]", completeDevice.MAC, completeDevice.ID))
+	c.logger.Debugf("registered mac address [%s] with id [%s] at [%s]:[%s]",
+		discoveredDevice.MAC,
+		discoveredDevice.ID,
+		newDevice.Host,
+		strconv.Itoa(newDevice.Port))
 
-	deviceWrapper := device.NewDeviceWrapper(*completeDevice)
-	err = deviceWrapper.RunHealthCheck(c.DeviceClient)
-	if err != nil {
-		return err
+	return *discoveredDevice, nil
+}
+
+// updateDeviceFromDevice builds a model.UpdateDevice from a model.Device.
+func updateDeviceFromDevice(d *model.Device) model.UpdateDevice {
```

Comment:
once we move away from graphql this should totally be on the model.Device struct